test/e2e/v2/plugin_cluster_test.go

```diff
@@ -217,11 +228,21 @@ var _ = Describe("kubebuilder", func() {
 			Expect(len(token)).To(BeNumerically(">", 0))
 
 			By("creating a pod with curl image")
-			cmdOpts := []string{
-				"run", "--generator=run-pod/v1", "curl", "--image=curlimages/curl:7.68.0", "--restart=OnFailure", "--",
-				"curl", "-v", "-k", "-H", fmt.Sprintf(`Authorization: Bearer %s`, token),
-				fmt.Sprintf("https://e2e-%v-controller-manager-metrics-service.e2e-%v-system.svc:8443/metrics",
-					kbc.TestSuffix, kbc.TestSuffix),
+			var cmdOpts []string
+			if clientVer := kbc.K8sVersion.ClientVersion; clientVer.GetMajorInt() <= 1 && clientVer.GetMinorInt() < 18 {
+				cmdOpts = []string{
+					"run", "--generator=run-pod/v1", "curl", "--image=curlimages/curl:7.68.0", "--restart=OnFailure", "--",
+					"curl", "-v", "-k", "-H", fmt.Sprintf(`Authorization: Bearer %s`, token),
+					fmt.Sprintf("https://e2e-%v-controller-manager-metrics-service.e2e-%v-system.svc:8443/metrics",
+						kbc.TestSuffix, kbc.TestSuffix),
+				}
+			} else {
+				cmdOpts = []string{
+					"run", "curl", "--image=curlimages/curl:7.68.0", "--restart=OnFailure", "--",
```

Comment:
If this works across all kubectl versions used, then you can get rid of the deprecated generator invocation completely.

Reply:
Well, prior to k8s 1.17 a pod would run with `--restart=Never` if generator[https://v1-17.docs.kubernetes.io/docs/reference/kubectl/conventions/#generators] not specified. Should leave it in place incase someone references Kubebuilder tools with a version of `kubectl` 1.17 or older?